docs/develop/smart-contracts/guidelines/message-delivery-guarantees.mdx

```diff
@@ -0,0 +1,70 @@
+
+If transaction in Ethereum is just a set of function calls in different contracts, transaction in TON is some inbound message that triggered it and some set of outbound messages which are sent to other contracts.
+
+## What is a Logical time?
+
+In such a system with asyncronous and parallel smart contract calls, it can be hard to define the order of actions to process. That's why each message in TON has it's _Logical time_ or _Lamport time_ (later just _lt_). It is used to understand which event caused another one and what do validator need to process first.
+
+It is strictly guaranteed that the transaction resulting from the message will have _lt_ greater than the _lt_ of the message. In order, _lt_ of message sent in some transaction is strictly greater than _lt_ of the transaction. Besides messages sent by one account as well as transactions happened on one account is strictly ordered as well. That way for each account we always know order of transactions, received and send messages.
+
+Moreover if account A sent two messages to account B, it is guaranteed that message with lower _lt_ will be processed earlier. Otherwise, an attempt to synchronize delivery would require the state of all the others to be known before processing one shard, and thereby break parallelization and destroy efficient sharding.
+
+
+For each block, we can define its _lt_ span, which starts with the first transaction and ends with the _lt_ of the last event in the block (message or transaction). Blocks are ordered the same way as any other events in TON, that way if one block depends on the other it has higher _lt_. Child block in shard has higher _lt_ than it's parent. Masterchain block _lt_ is higher that _lt's_ of shard blocks it lists, since master block _depends_ on listed shardblocks. In order each shard block contains reference to latest (at the moment of shard block creation) masterblock and thus shard block _lt_ is higher than referenced master block _lt_.
+
+## Message delivery
+
+Fortunately, TON is working such way that any internal message will surely be received on the destination account. Message cannot be lost anywhere between the source and destination. External messages is a little bit different since it's acceptance to the block is at validators discretionm however once message is accepted to the incoming message queue it will be delivered.
+
+### Delivery order
+
+It looks like _lt_ solves the issue about message delivery order, because we know that transaction with lower _lt_ will be processed first. But doesn't work for all scenarios.
+
+Suppose that there are two contracts - _A_ and _B_. Then _A_ receives an external message which triggers it to send two internal messages to _B_, let's call these messages _1_ and _2_. In that simple case we can be 100% sure that _1_ will be processed by _B_ before _2_ because it has lower _lt_.
```

Comment:
Suppose that there are two contracts - _A_ and _B_. _A_ receives an external message which triggers it to send two internal messages to _B_, let's call these messages _1_ and _2_. In **this** simple case, we can be 100% sure that _1_ will be processed by _B_ before _2_ because it has **a** lower _lt_.